python/modem-firmware-1.3+/nrf_cloud_device_mgmt.py

```diff
@@ -0,0 +1,717 @@
+
+def get_device_list(api_key, fota_types_list, device_id):
+    hdr = {'Authorization': 'Bearer ' + api_key}
+    req_base = GET_DEVICES_BASE + '&pageLimit={}'.format(GET_DEVICES_PAGE_LIMIT)
+
+    if fota_types_list:
+        for type in fota_types_list:
+            req_base = req_base + '&firmwareSupport={}'.format(type)
+
+    if device_id:
+        req_base = req_base + '&deviceIds={}'.format(device_id)
+
+    next_tok = ''
+    dev_list = []
+    total = 0
+
+    while True:
+
+        req = req_base
+
+        if next_tok:
+            req = req + '&pageNextToken={}'.format(next_tok)
+
+        api_res = requests.get(req, headers=hdr)
+
+        if api_res.status_code != 200:
+            print_api_result("ListDevices API call failed", api_res, True)
+            break
+
+        api_res_json = api_res.json()
+
+        if 'items' in api_res_json.keys():
+            items = api_res_json['items']
+            for i in items:
+                dev_list.append(nRFCloudDevice(i))
+
+        if 'total' in api_res_json.keys():
+            total = api_res_json['total']
+
+        if 'pageNextToken' in api_res_json.keys():
+            next_tok = api_res_json['pageNextToken']
+        else:
+            next_tok = ''
+            break
+
+    return sorted(dev_list,  key=lambda dev : dev.id)
+
+def print_api_result(custom_text, api_result, print_response_txt):
+    print("{}: {} - {}".format(custom_text, api_result.status_code, api_result.reason))
+    if print_response_txt:
+        print("Response: {}".format(api_result.text))
+
+def create_fota_job(api_key, json_payload_obj):
+    jobId = None
+    hdr = {'Authorization': 'Bearer ' + api_key}
+    req = FOTA_JOBS_BASE
+
+    api_res = requests.post(req, json=json_payload_obj, headers=hdr)
+    if api_res.status_code != 200:
+            print_api_result("ListFirmware API call failed", api_res, True)
+    else:
+        api_res_json = api_res.json()
+
+        if 'jobId' in api_res_json.keys():
+            jobId = api_res_json['jobId']
+
+    return jobId
+
+def get_fota_job_payload_common(bundle_id, update_name, update_desc, apply):
+    payload = {}
+    payload['bundleId'] = bundle_id
+    if not apply:
+        # param defaults to "true"
+        payload['autoApply'] = "false"
+    payload['name'] = update_name
+    payload['description'] = update_desc
+    return payload
+
+def create_fota_job_by_tag(api_key, bundle_id, update_name, update_desc, tag_list, apply):
+
+    payload = get_fota_job_payload_common(bundle_id, update_name, update_desc, apply)
+    payload['tags'] = tag_list
+
+    return create_fota_job(api_key, payload)
+
+def create_fota_job_by_device_id(api_key, bundle_id, update_name, update_desc, device_id_list, apply):
+
+    payload = get_fota_job_payload_common(bundle_id, update_name, update_desc, apply)
+    payload['deviceIds'] = device_id_list
+
+    return create_fota_job(api_key, payload)
+
+def user_select_from_list(the_list):
```

Comment:
Since you don't use the contents of the list, just the size of it, perhaps the size should be the parameter.